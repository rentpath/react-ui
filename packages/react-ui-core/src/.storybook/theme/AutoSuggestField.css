.AutoSuggestField {
  display: flex;
  margin: 75px;
}

.AutoSuggestField .Dropdown,
.AutoSuggestField .Field {
  margin: 0;
  width: 300px;
}

.AutoSuggestField .Button {
  height: 40px;
  position: relative;
  left: -70px;
  top: 5px;
}

.AutoSuggestField .Card {
  height: auto;
  width: auto;
  text-align: left;
}

.AutoSuggestField .List {
  padding: 0;
}

.AutoSuggestField .ListItem {
  padding: 5px 0;
}

.AutoSuggestField .Highlighter-Match {
  background: none;
}

.AutoSuggestField .Menu {
  padding-left: 5px;
}

.clear .Button {
  left: -40px;
}
