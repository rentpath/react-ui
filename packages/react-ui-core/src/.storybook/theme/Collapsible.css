.Collapsible_Items-visible {
  display: block;
}

.Collapsible_Items-hidden {
  display: none;
}

.Collapsible .Label {
  min-width: 200px;
}
