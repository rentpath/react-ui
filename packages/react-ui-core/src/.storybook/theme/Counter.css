.Counter {
  display: flex;
  white-space: nowrap;
  padding: 10px;
  align-items: center;
}

.Counter .Label {
  display: inline-block;
  white-space: nowrap;
  padding-right: 10px;
}

.Counter_Controls {
  display: inline-block;
  white-space: nowrap;
}

.Counter .Text {
  display: inline-block;
  text-align: center;
  padding: 0 20px;
}

.Counter_Button span {
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
  display: inline-block;
  background: #0175D7;
  cursor: pointer;
}

.Counter_Button .Button {
  border-radius: 50%;
  width: 38px;
}

.Counter .Button {
  color: #ffffff;
  font-weight: bold;
}
