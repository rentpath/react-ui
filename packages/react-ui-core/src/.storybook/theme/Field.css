.Input,
.Select,
.Textarea {
  background-color: #ffffff;
  box-shadow: 0 0.5px 1px 0 rgba(0, 0, 0, 0.08);
  border: solid 1px #e9e9e9;
  border-radius: 4px;
  padding: 16px 10px;
  font-size: 13px;
}

.Field {
  margin: 16px 0;
  display: flex;
  max-width: 600px;
  color: #000;
  align-items: center;
}

.Field .Label {
  padding-right: 10px;
}
