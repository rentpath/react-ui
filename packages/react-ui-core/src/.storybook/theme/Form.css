.Form {
  min-width: 500px;
}

.Form .Field {
  color:#000;
}

.Form .Text {
  text-transform: uppercase;
  padding-bottom: 10px;
  margin-bottom: 10px;
}

.Form .Label {
  flex: 1;
}

.Form .Input,
.Form .Select,
.Form .Textarea {
  flex: 4;
}
