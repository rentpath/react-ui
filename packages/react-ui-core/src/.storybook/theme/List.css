.List {
  display: flex;
  padding: 20px;
}

.List-horizontal {
  flex-direction: row;
  justify-content: space-around;
}

.List-horizontal .ListItem {
  width: 100px;
}

.List-vertical {
  flex-direction: column;
}

.ListItem {
  cursor: pointer;
}

.ListItem-highlight {
  color: #AAA;
}
