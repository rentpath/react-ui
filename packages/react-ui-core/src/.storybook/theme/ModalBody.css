.ModalBody {
  min-width: 300px;
  padding: 20px;
  background: #FFF;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 1px 1px 5px #666;
  position: relative;
}
