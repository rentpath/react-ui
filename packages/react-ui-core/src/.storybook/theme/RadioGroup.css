.RadioGroup {
  border-radius: 5px;
  padding: .5rem 1rem;
}
