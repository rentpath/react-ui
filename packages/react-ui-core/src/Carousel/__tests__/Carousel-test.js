import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import LazyLoad from 'react-lazyload'
import theme from './mocks/theme'
import ThemedCarousel from '../Carousel'
import ThemedCarouselNavigation from '../CarouselNavigation'

const Carousel = ThemedCarousel.WrappedComponent
const Navigation = ThemedCarouselNavigation.WrappedComponent

const onClick = jest.fn()

const items = [
  <div key="1">Test 1</div>,
  <div key="2">Test 2</div>,
  <div key="3">Test 3</div>,
  <div key="4">Test 4</div>,
]

describe('Carousel', () => {
  it('applies a base theme to the root node', () => {
    const wrapper = shallow(<Carousel theme={theme} />)
    const className = wrapper.prop('className')
    expect(className).toEqual('Carousel')
  })

  it('applies a className to the root node', () => {
    const wrapper = shallow(<Carousel theme={theme} className="Foo" />)
    const className = wrapper.prop('className')
    expect(className).toContain('Foo')
  })

  it('includes a static data-tid', () => {
    const wrapper = shallow(<Carousel />)
    expect(wrapper.find('[data-tid="carousel"]')).toHaveLength(1)
  })

  it('renders using a custom renderItem prop', () => {
    const wrapper = mount(
      <Carousel
        items={items}
        renderItem={item => <div className="TestItem">{item}</div>}
      />
    )
    expect(wrapper.find('.TestItem')).toHaveLength(4)
  })

  it('lazyLoads images if required', () => {
    const wrapper = shallow(
      <Carousel
        theme={theme}
        items={items}
        lazyLoad
      />
    )

    expect(wrapper.find(<LazyLoad />)).toBeTruthy()
  })

  describe('slideToIndex', () => {
    let wrapper
    let slide

    beforeEach(() => {
      wrapper = shallow(<Carousel items={items} />)
      slide = jest.fn()
    })

    it('slides to index when carousel and index exist', () => {
      wrapper.instance().carousel = { slideToIndex: slide }
      wrapper.instance().slideToIndex(1)
      expect(slide).toHaveBeenCalled()
    })

    it('does not slide to index when index does not exist', () => {
      wrapper.instance().carousel = { slideToIndex: slide }
      wrapper.instance().slideToIndex(100)
      expect(slide).not.toHaveBeenCalled()
    })

    it('does not slide to index when carousel does not exist', () => {
      wrapper.instance().carousel = undefined
      wrapper.instance().slideToIndex(100)
      expect(slide).not.toHaveBeenCalled()
    })
  })

  describe('navigation', () => {
    it('does not render navigation if none passed', () => {
      const snap = renderer.create(<Carousel items={items} />).toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('renders navigation buttons based on object', () => {
      const snap = renderer.create(
        <Carousel
          navigation={{
            next: {
              children: 'Next',
            },
            previous: {
              children: 'Previous',
            },
          }}
          items={items}
        />
      )
      .toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('renders navigation buttons based on custom node', () => {
      const snap = renderer.create(
        <Carousel
          navigation={{
            previous: (<Navigation key="pior" direction="previous">Prior</Navigation>),
            next: (<Navigation key="upcoming" direction="next">Upcoming</Navigation>),
          }}
          items={items}
        />
      ).toJSON()

      expect(snap).toMatchSnapshot()
    })
  })

  describe('pagination', () => {
    const paginationProps = {
      pageNumber: 1,
      total: 10,
      pageSize: 2,
      next: {
        children: 'Next Page',
        className: 'someName',
      },
      previous: {
        children: 'Previous Page',
        className: 'anotherName',
      },
    }

    it('displays next card when there are more listings to view', () => {
      const wrapper = mount(<Carousel pagination={paginationProps} items={items} />)
      const previous = wrapper.find('button[data-tid="carousel-pagination-previous"]')
      const next = wrapper.find('button[data-tid="carousel-pagination-next"]')

      expect(previous).toHaveLength(0)
      expect(next.text()).toEqual('Next Page')
    })

    it('displays previous and next card when not on the first page', () => {
      const pagination = { ...paginationProps, pageNumber: 2 }
      const wrapper = mount(<Carousel pagination={pagination} items={items} />)
      const previous = wrapper.find('button[data-tid="carousel-pagination-previous"]')
      const next = wrapper.find('button[data-tid="carousel-pagination-next"]')
      expect(previous.text()).toEqual('Previous Page')
      expect(next.text()).toEqual('Next Page')
    })

    it('displays previous card only when not on first page and total does not exceed current page', () => {
      const pagination = { ...paginationProps, pageNumber: 4 }
      const wrapper = shallow(<Carousel pagination={pagination} items={items} />)
      const renderedItems = wrapper.find('ImageGallery').props().items
      expect(renderedItems[0].props.children).toEqual('Previous Page')
      expect(renderedItems[renderedItems.length - 1].props.children).toEqual('Next Page')
    })

    it('does not call `onClick` when none passed', () => {
      const props = { ...paginationProps, pageNumber: 2 }
      const wrapper = mount(<Carousel pagination={props} items={items} />)
      const previous = wrapper.find('button[data-tid="carousel-pagination-previous"]')
      const next = wrapper.find('button[data-tid="carousel-pagination-next"]')
      previous.simulate('click')
      next.simulate('click')
      expect(onClick).not.toHaveBeenCalled()
    })

    it('calls `onClick` when passed', () => {
      const props = {
        pageNumber: 2,
        total: 10,
        pageSize: 2,
        next: {
          id: 'next',
          onClick,
        },
        previous: {
          id: 'previous',
          onClick,
        },
      }

      const wrapper = mount(<Carousel pagination={props} items={items} />)
      const previous = wrapper.find('button[data-tid="carousel-pagination-previous"]')
      const next = wrapper.find('button[data-tid="carousel-pagination-next"]')
      previous.simulate('click')
      next.simulate('click')
      expect(onClick).toHaveBeenCalled()
      expect(onClick).toHaveBeenCalled()
    })
  })
})
