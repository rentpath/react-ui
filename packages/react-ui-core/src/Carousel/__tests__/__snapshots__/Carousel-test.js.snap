// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel navigation does not render navigation if none passed 1`] = `
<div
  className=""
  data-tid="carousel"
>
  <div
    aria-live="polite"
    className="image-gallery "
  >
    <div
      className="image-gallery-content"
    >
      <div
        className="image-gallery-slide-wrapper bottom"
      >
        <div
          className="image-gallery-swipe"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="image-gallery-slides"
          >
            <div
              className="image-gallery-slide center"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(0%, 0, 0)",
                  "OTransform": "translate3d(0%, 0, 0)",
                  "WebkitTransform": "translate3d(0%, 0, 0)",
                  "msTransform": "translate3d(0%, 0, 0)",
                  "transform": "translate3d(0%, 0, 0)",
                }
              }
            >
              <div>
                Test 1
              </div>
            </div>
            <div
              className="image-gallery-slide right"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(100%, 0, 0)",
                  "OTransform": "translate3d(100%, 0, 0)",
                  "WebkitTransform": "translate3d(100%, 0, 0)",
                  "msTransform": "translate3d(100%, 0, 0)",
                  "transform": "translate3d(100%, 0, 0)",
                }
              }
            >
              <div>
                Test 2
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(200%, 0, 0)",
                  "OTransform": "translate3d(200%, 0, 0)",
                  "WebkitTransform": "translate3d(200%, 0, 0)",
                  "msTransform": "translate3d(200%, 0, 0)",
                  "transform": "translate3d(200%, 0, 0)",
                }
              }
            >
              <div>
                Test 3
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(300%, 0, 0)",
                  "OTransform": "translate3d(300%, 0, 0)",
                  "WebkitTransform": "translate3d(300%, 0, 0)",
                  "msTransform": "translate3d(300%, 0, 0)",
                  "transform": "translate3d(300%, 0, 0)",
                }
              }
            >
              <div>
                Test 4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel navigation renders navigation buttons based on custom node 1`] = `
<div
  className=""
  data-tid="carousel"
>
  <div
    aria-live="polite"
    className="image-gallery "
  >
    <div
      className="image-gallery-content"
    >
      <div
        className="image-gallery-slide-wrapper bottom"
      >
        <span>
          <button
            $$typeof={Symbol(react.element)}
            _owner={null}
            _store={Object {}}
            className=""
            data-tid="carousel-navigation-previous"
            disabled={true}
            onClick={[Function]}
            props={
              Object {
                "children": "Prior",
                "direction": "previous",
                "theme": Object {},
              }
            }
            role="button"
            tabIndex={0}
            type={[Function]}
          />
          <button
            $$typeof={Symbol(react.element)}
            _owner={null}
            _store={Object {}}
            className=""
            data-tid="carousel-navigation-next"
            disabled={false}
            onClick={[Function]}
            props={
              Object {
                "children": "Upcoming",
                "direction": "next",
                "theme": Object {},
              }
            }
            role="button"
            tabIndex={0}
            type={[Function]}
          />
        </span>
        <div
          className="image-gallery-swipe"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="image-gallery-slides"
          >
            <div
              className="image-gallery-slide center"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(0%, 0, 0)",
                  "OTransform": "translate3d(0%, 0, 0)",
                  "WebkitTransform": "translate3d(0%, 0, 0)",
                  "msTransform": "translate3d(0%, 0, 0)",
                  "transform": "translate3d(0%, 0, 0)",
                }
              }
            >
              <div>
                Test 1
              </div>
            </div>
            <div
              className="image-gallery-slide right"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(100%, 0, 0)",
                  "OTransform": "translate3d(100%, 0, 0)",
                  "WebkitTransform": "translate3d(100%, 0, 0)",
                  "msTransform": "translate3d(100%, 0, 0)",
                  "transform": "translate3d(100%, 0, 0)",
                }
              }
            >
              <div>
                Test 2
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(200%, 0, 0)",
                  "OTransform": "translate3d(200%, 0, 0)",
                  "WebkitTransform": "translate3d(200%, 0, 0)",
                  "msTransform": "translate3d(200%, 0, 0)",
                  "transform": "translate3d(200%, 0, 0)",
                }
              }
            >
              <div>
                Test 3
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(300%, 0, 0)",
                  "OTransform": "translate3d(300%, 0, 0)",
                  "WebkitTransform": "translate3d(300%, 0, 0)",
                  "msTransform": "translate3d(300%, 0, 0)",
                  "transform": "translate3d(300%, 0, 0)",
                }
              }
            >
              <div>
                Test 4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel navigation renders navigation buttons based on object 1`] = `
<div
  className=""
  data-tid="carousel"
>
  <div
    aria-live="polite"
    className="image-gallery "
  >
    <div
      className="image-gallery-content"
    >
      <div
        className="image-gallery-slide-wrapper bottom"
      >
        <span>
          <button
            className=""
            data-tid="carousel-navigation-previous"
            disabled={true}
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Previous
          </button>
          <button
            className=""
            data-tid="carousel-navigation-next"
            disabled={false}
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Next
          </button>
        </span>
        <div
          className="image-gallery-swipe"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="image-gallery-slides"
          >
            <div
              className="image-gallery-slide center"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(0%, 0, 0)",
                  "OTransform": "translate3d(0%, 0, 0)",
                  "WebkitTransform": "translate3d(0%, 0, 0)",
                  "msTransform": "translate3d(0%, 0, 0)",
                  "transform": "translate3d(0%, 0, 0)",
                }
              }
            >
              <div>
                Test 1
              </div>
            </div>
            <div
              className="image-gallery-slide right"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(100%, 0, 0)",
                  "OTransform": "translate3d(100%, 0, 0)",
                  "WebkitTransform": "translate3d(100%, 0, 0)",
                  "msTransform": "translate3d(100%, 0, 0)",
                  "transform": "translate3d(100%, 0, 0)",
                }
              }
            >
              <div>
                Test 2
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(200%, 0, 0)",
                  "OTransform": "translate3d(200%, 0, 0)",
                  "WebkitTransform": "translate3d(200%, 0, 0)",
                  "msTransform": "translate3d(200%, 0, 0)",
                  "transform": "translate3d(200%, 0, 0)",
                }
              }
            >
              <div>
                Test 3
              </div>
            </div>
            <div
              className="image-gallery-slide"
              onClick={undefined}
              onMouseLeave={undefined}
              onMouseOver={undefined}
              onTouchEnd={undefined}
              onTouchMove={undefined}
              onTouchStart={undefined}
              role={undefined}
              style={
                Object {
                  "MozTransform": "translate3d(300%, 0, 0)",
                  "OTransform": "translate3d(300%, 0, 0)",
                  "WebkitTransform": "translate3d(300%, 0, 0)",
                  "msTransform": "translate3d(300%, 0, 0)",
                  "transform": "translate3d(300%, 0, 0)",
                }
              }
            >
              <div>
                Test 4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
